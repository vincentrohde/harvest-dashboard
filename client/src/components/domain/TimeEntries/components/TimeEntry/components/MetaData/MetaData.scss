.meta-data-container {
    position: relative;
    flex: 1;

    .title {
        position: absolute;
        bottom: 0;
        font-weight: 400;
        font-size: 18px;
        margin: 10px 0 0;

        @media (max-width:600px)  {
            position: relative;
            bottom: unset;
        }
    }

    .meta-data {
        font-size: 14px;

        span {
            &.group {
                color: dodgerblue;
            }
        }
    }
}
