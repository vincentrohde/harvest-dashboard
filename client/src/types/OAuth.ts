export type oAuthType = {
    access_token: string;
    account_id: string;
    expiration_date: string;
    refresh_token: string;
};
