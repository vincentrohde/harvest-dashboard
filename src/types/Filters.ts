export interface FiltersInterface {
    dateRange: string[] | 1;
}
