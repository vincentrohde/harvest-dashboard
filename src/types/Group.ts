export type groupsType = GroupInterface[];

export interface GroupInterface {
    id: number;
    name: string;
}