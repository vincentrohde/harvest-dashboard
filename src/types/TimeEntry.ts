import { ProjectInterface } from './Project';
import { TaskInterface } from './Task';

export type timeEntriesType = TimeEntryInterface[];

export interface ShallowTimeEntryInterface {
    hours: number | string;
    id: number;
    notes: string;
    spent_date: string;
    is_running: boolean;
    project: ProjectInterface;
    task: TaskInterface;
}

export interface TimeEntryInterface extends ShallowTimeEntryInterface {
    hours: number;
}

export interface ShallowSubmissionEntryInterface {
    notes: ShallowTimeEntryInterface['notes'];
    spent_date: ShallowTimeEntryInterface['spent_date'];
    hours: ShallowTimeEntryInterface['hours'];
    project_id: number | string;
    task_id: number | string;
}

export interface TimeEntrySubmissionInterface extends ShallowSubmissionEntryInterface {
    hours: TimeEntryInterface['hours'];
    project_id: number;
    task_id: number;
}