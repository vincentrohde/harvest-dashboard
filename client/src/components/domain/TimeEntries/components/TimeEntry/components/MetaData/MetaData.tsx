// Libs
import React from 'react';

// Types
import { MetaDataProps } from './MetaData.types';

// Styles
import './MetaData.scss';

const MetaData = ({notes, date, task}: MetaDataProps) => (<div className="meta-data-container">
    <p className="meta-data pipes">
        <span className="category">{task}</span>
        <span>{date}</span>
    </p>
    <h3 className="title">{notes}</h3>
</div>);

export default MetaData;
