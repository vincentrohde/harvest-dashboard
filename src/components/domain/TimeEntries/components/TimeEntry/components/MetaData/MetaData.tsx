// Libs
import React from 'react';
import classNames from 'classnames';

// Types
import { MetaDataProps } from './MetaData.types';

// Styles
import styles from './MetaData.module.scss';

const MetaData = ({notes, date, task}: MetaDataProps) => (<div className={styles['meta-data-container']}>
    <p className={classNames(styles['meta-data'], 'pipes')}>
        <span className={styles.task}>{task}</span>
        <span>{date}</span>
    </p>
    <h3 className={styles.title}>{notes}</h3>
</div>);

export default MetaData;
