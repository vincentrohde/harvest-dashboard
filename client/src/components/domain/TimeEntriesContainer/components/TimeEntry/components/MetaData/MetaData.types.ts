import { TimeEntryInterface } from '@/types/TimeEntry';
import { TaskInterface } from '@/types/Task';

export interface MetaDataProps {
    date: TimeEntryInterface['spent_date'];
    notes: TimeEntryInterface['notes'];
    task: TaskInterface['name'];
}
