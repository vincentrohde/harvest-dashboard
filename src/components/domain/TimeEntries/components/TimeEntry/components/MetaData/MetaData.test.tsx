// Libs
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// Components
import MetaData from './MetaData';

// Types
import { MetaDataProps } from './MetaData.types';

const mockData: MetaDataProps = {
    date: '01.01.2022',
    notes: 'Test notes',
    task: 'Test task'
}

describe('<MetaData />', () => {
    it('renders', () => {
        const { asFragment } = render(
            <MetaData {...mockData} />
        );

        expect(asFragment()).toMatchSnapshot();
    });
})
