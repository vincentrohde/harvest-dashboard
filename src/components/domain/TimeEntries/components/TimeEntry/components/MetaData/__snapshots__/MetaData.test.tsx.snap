// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetaData /> renders 1`] = `
<DocumentFragment>
  <div
    class="meta-data-container"
  >
    <p
      class="meta-data pipes"
    >
      <span
        class="task"
      >
        Test task
      </span>
      <span>
        01.01.2022
      </span>
    </p>
    <h3
      class="title"
    >
      Test notes
    </h3>
  </div>
</DocumentFragment>
`;
